### DISCUSSION TITLE
[RESOLVED] References to old repo (github.com/docker/docker/*)

### DISCUSSION THREAD

**Discussion by SercioSoydanov** [moby/moby#44786] [General]
There are still references to the old repo url (github.com/docker/docker/) in the code which causes problems when I'm trying to use the libraries for a project of mine. 

Is there a specific reason they are left the way they are, or is it just because nobody has fixed it yet :)

**Comment by tao12345666333** [comment]
In fact there are automatic redirects

**Reply by SercioSoydanov** [reply]
Automatic redirect works domain-wise, but the problem arises when I'm trying to use the individual modules from within another project.

**Comment by thaJeztah** [comment]
This is mostly on purpose; while the repository has been renamed to `github.com/moby/moby`, the package itself has not; it's still named `github.com/docker/docker`, and must be imported as such (the [`//import ..`](https://github.com/moby/moby/blob/219ced474d8bcb9cbd399a6ab6ec17f178fe3306/pkg/archive/archive.go#L2) comments in various locations are to enforce that, although they may be ignored with go modules enabled).

This repository currently does not yet provide a go module (one of the reasons for this was that tags on this repository are for _binary_ releases, and the use of CalVer for those tags). We do plan to migrate to go modules, but there's still a significant amount of work (and planning / coordination between other projects) to be done before we're able to. When we do fully move to go modules, we will also rename the package to `github.com/moby/moby`, which effectively means it's a new module that didn't exist before.

**Comment by SercioSoydanov** [comment]
@thaJeztah Thanks a lot for the explanation. I'll be looking forward for migration to go modules and until then I'll just manually adjust the imported references locally.

**Comment by thaJeztah** [comment]
@SercioSoydanov can you explain the problem a bit more? It should not be needed to replace the name, as long as you use the correct name for the package / module (currently `github.com/docker/docker`), things should work as expected; don't use `github.com/moby/moby` because that's not the correct name for the module/package. See https://github.com/moby/buildkit/blob/7bd72856ad4d403260a6f68c31977c72c2d1d8ce/go.mod#L30 for example, how it's used in BuildKit

**Reply by SercioSoydanov** [reply]
@thaJeztah Well as it turns out, it was totally my mistake. I've included the old repo to the project **before** it's been moved to `/moby/moby`. Then I've tried to retidy the project, but it failed to resolve the dependency (probably beacuse old hash was written in the `go.sum` file) and suddenly it stopped working. 

Deleting the `go.mod` and `go.sum` ad re initializing / tidying them with proper package name as you described seems to solve it. Thanks a lot for your support.

**Reply by thaJeztah** [reply]
Good to hear!

I definitely agree it's very confusing, and we want to get out of the confusing situation, but (as mentioned) a lot of work is involved, and we need to make changes _very_ carefully as there's many things / projects / consumers depending on this, so we're doing it in small steps.

**Reply by SercioSoydanov** [reply]
I had a futile attempt to modify the all references to new repo name, so I can imagine the volume of the effort that is involved. Also considering the number of installments that depend on this project, I don't think that you can be too careful with any change you plan to make. So that totally makes sense. 

Thanks for all the hard work & beautiful piece of software. Keep it up!
